model_analyzer/result/constraint_manager.py

```diff
@@ -61,13 +62,15 @@ def check_constraints(constraints, measurement):
         """
 
         if constraints:
-            for metric in measurement.data():
-                if type(metric).tag in constraints:
-                    constraint = constraints[type(metric).tag]
-                    if 'min' in constraint:
-                        if metric.value() < constraint['min']:
-                            return False
-                    if 'max' in constraint:
-                        if metric.value() > constraint['max']:
-                            return False
+            for model_metrics in run_config_measurement.non_gpu_data():
```

Comment:
do constraints only apply to non-gpu data?

Reply:
Yes, that is my understanding and I don't see why multi-model would want to change that. The GPU metrics are just informational: utilization and memory usage

Reply:
We talked on the phone and confirmed that this does need to be fixed -- we do support gpu memory as a constraint

Reply:
I've changed this back to be data().

Reply:
I've added a check for gpu memory used in the report manager unit testing